tests/test_dataset_broker.py

```diff
@@ -7,37 +7,66 @@
 
 import pytest
 import os.path
+import re
 import time
-from subprocess import Popen
-from subprocess import call
+import tempfile
 import signal
 
+from subprocess import call, Popen
+
 producer_path = "../build/tests/boost/dataset_broker_producer"
 producer2_path = "../build/tests/boost/dataset_broker_producer2"
 consumer_path = "../build/tests/boost/dataset_broker_consumer"
-broker_path = "../build/ext/src/ch_acq/dataset_broker.py"
 
 
 def test_produce_consume():
+    broker_path = [
+        "/usr/local/bin/comet",
+        os.path.join(os.path.expanduser("~"), ".local/bin/comet"),
+    ]
     if (
         not os.path.isfile(producer_path)
         or not os.path.isfile(consumer_path)
         or not os.path.isfile(producer2_path)
     ):
-        print("Deactivated! Build with -DBOOST_TESTS=ON to activate this test")
-        return
-    if not os.path.isfile(broker_path):
-        print("Deactivated! Make sure the dataset_broker is at {}".format(broker_path))
-        return
-
-    broker = Popen([broker_path])
-    time.sleep(2)
-
-    try:
-        assert call([producer_path]) == 0
-        assert call([producer2_path]) == 0
-        assert call([consumer_path]) == 0
-    finally:
-        pid = broker.pid
-        os.kill(pid, signal.SIGINT)
-        broker.terminate()
+        pytest.skip("Build with -DBOOST_TESTS=ON to activate this test")
+    if not os.path.isfile(broker_path[0]):
+        if not os.path.isfile(broker_path[1]):
+            pytest.skip("Make sure the dataset_broker is at {}".format(broker_path))
+        broker_path = broker_path[1]
+    else:
+        broker_path = broker_path[0]
+
+    with tempfile.NamedTemporaryFile(mode="w") as f_out:
+        # Start comet with a random port
+        broker = Popen(
+            [broker_path, "-p", "0", "--recover", "False"], stdout=f_out, stderr=f_out
+        )
+        time.sleep(3)
+
+        # Find port in the log
```

Comment:
Ugh.

Can't you just pick a random port in this code and then set the port when you run the broker?

Reply:
The tests run in parallel on the CI server. So they could pick the same random free port in parallel tests and mysteriously fail.